File: app/models/watch_digest.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field

from app.utils.timezone import now_tz


class WatchRule(BaseModel):
    user_id: str
    stock_code: str
    stock_name: Optional[str] = None
    market: str = "A股"
    schedule_type: str = "daily_post_market"
    cron_expr: Optional[str] = None
    status: str = "active"
    created_at: datetime = Field(default_factory=now_tz)
    updated_at: datetime = Field(default_factory=now_tz)


class WatchDigest(BaseModel):
    user_id: str
    stock_code: str
    stock_name: Optional[str] = None
    market: str = "A股"
    report_id: Optional[str] = None
    task_id: Optional[str] = None
    summary: str = ""
    recommendation: Optional[str] = None
    risk_level: str = "中等"
    confidence_score: Optional[float] = None
    status: str = "ready"
    generated_at: datetime = Field(default_factory=now_tz)
    updated_at: datetime = Field(default_factory=now_tz)


class WatchDigestCard(BaseModel):
    stock_code: str
    stock_name: str
    market: str = "A股"
    board: Optional[str] = None
    exchange: Optional[str] = None
    current_price: Optional[float] = None
    change_percent: Optional[float] = None
    summary: str = ""
    recommendation: Optional[str] = None
    risk_level: str = "未配置"
    confidence_score: Optional[float] = None
    schedule_type: Optional[str] = None
    rule_status: str = "inactive"
    updated_at: Optional[str] = None
    report_id: Optional[str] = None
    task_id: Optional[str] = None

File: frontend/src/api/watchDigest.ts
import { ApiClient } from './request'

export interface WatchDigestRule {
  stock_code: string
  stock_name?: string
  market?: string
  schedule_type: string
  schedule_label?: string
  cron_expr?: string | null
  status: string
  updated_at?: string | null
}

export interface WatchDigestCard {
  stock_code: string
  stock_name: string
  market: string
  board?: string | null
  exchange?: string | null
  current_price?: number | null
  change_percent?: number | null
  summary: string
  recommendation?: string | null
  risk_level: string
  confidence_score?: number | null
  schedule_type?: string | null
  schedule_label?: string
  rule_status: string
  updated_at?: string | null
  report_id?: string | null
  task_id?: string | null
}

export const watchDigestApi = {
  list: () => ApiClient.get<WatchDigestCard[]>('/api/watch/digests'),

  listRules: () => ApiClient.get<WatchDigestRule[]>('/api/watch/rules'),

  saveRule: (
    stockCode: string,
    payload: {
      stock_name?: string
      market?: string
      schedule_type: string
      cron_expr?: string | null
      status: string
    }
  ) => ApiClient.put<WatchDigestRule>(`/api/watch/rules/${stockCode}`, payload),

  deleteRule: (stockCode: string) => ApiClient.delete<{ stock_code: string }>(`/api/watch/rules/${stockCode}`),

  refreshOne: (stockCode: string, payload: { stock_name: string; market?: string }) =>
    ApiClient.post<{ task_id: string; status: string; message: string }>(`/api/watch/digests/${stockCode}/refresh`, payload),

  refreshAll: () => ApiClient.post<{ count: number; stocks: string[] }>('/api/watch/digests/refresh-all'),
}
